<a href="https://colab.research.google.com/github/Sandyjiit/Machine-learning/blob/main/Breast_Cancer_Classification_by_SVM.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#libariries imported
import numpy as np
import pandas as pd


In [2]:
#loading dataset
df_cancr=pd.read_csv("/content/Breast_cancer_data.csv")


In [3]:
df_cancr.head(5)

mean_radius  mean_texture  ...  mean_smoothness  diagnosis
0        17.99         10.38  ...          0.11840          0
1        20.57         17.77  ...          0.08474          0
2        19.69         21.25  ...          0.10960          0
3        11.42         20.38  ...          0.14250          0
4        20.29         14.34  ...          0.10030          0

[5 rows x 6 columns]

In [4]:
df_cancr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   mean_radius      569 non-null    float64
 1   mean_texture     569 non-null    float64
 2   mean_perimeter   569 non-null    float64
 3   mean_area        569 non-null    float64
 4   mean_smoothness  569 non-null    float64
 5   diagnosis        569 non-null    int64  
dtypes: float64(5), int64(1)
memory usage: 26.8 KB


As We can observe the dataet,there is absence for non-null value,so data is ready to feed forward for ML model prepossing steps.

In [5]:
df_cancr.describe()


mean_radius  mean_texture  ...  mean_smoothness   diagnosis
count   569.000000    569.000000  ...       569.000000  569.000000
mean     14.127292     19.289649  ...         0.096360    0.627417
std       3.524049      4.301036  ...         0.014064    0.483918
min       6.981000      9.710000  ...         0.052630    0.000000
25%      11.700000     16.170000  ...         0.086370    0.000000
50%      13.370000     18.840000  ...         0.095870    1.000000
75%      15.780000     21.800000  ...         0.105300    1.000000
max      28.110000     39.280000  ...         0.163400    1.000000

[8 rows x 6 columns]

We can see the stastical properties of dataset.

**Now,we need to spilt dataset for dependent and independent variables for traing and testing.**

In [6]:
#By droping the  diaognosis column as it is dependent on the rest of the other parameter for the identification of type of cancer.

In [8]:
X=df_cancr.drop(['diagnosis'],axis=1).values
X

array([[1.799e+01, 1.038e+01, 1.228e+02, 1.001e+03, 1.184e-01],
       [2.057e+01, 1.777e+01, 1.329e+02, 1.326e+03, 8.474e-02],
       [1.969e+01, 2.125e+01, 1.300e+02, 1.203e+03, 1.096e-01],
       ...,
       [1.660e+01, 2.808e+01, 1.083e+02, 8.581e+02, 8.455e-02],
       [2.060e+01, 2.933e+01, 1.401e+02, 1.265e+03, 1.178e-01],
       [7.760e+00, 2.454e+01, 4.792e+01, 1.810e+02, 5.263e-02]])

In [10]:
Y=df_cancr['diagnosis'].values
Y

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0,
       1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0,
       1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1,
       1, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0,
       1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1,
       1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0,

*Data is being splitted for test_train split with 7:3 ration**

In [25]:
from sklearn.model_selection import train_test_split
from sklearn import metrics

In [36]:
X_train, X_test, Y_train, Y_test=train_test_split(X,Y,test_size=.2,random_state=51)

In [37]:
Y_predict=svc_model.predict(X_test) #model is predicting on test data

**MODEL TRAINING**

In [38]:
from sklearn.svm import SVC
svc_model=SVC()
svc_model.fit(X_train,Y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='rbf',
    max_iter=-1, probability=False, random_state=None, shrinking=True,
    tol=0.001, verbose=False)

**FOR MODEL UNDERSTAING CLASSIFICATION REPORT AND CONFUSION MATRIX**

In [39]:
from sklearn.metrics import  classification_report,confusion_matrix
cm=confusion_matrix(Y_test,Y_predict)
cm

array([[27, 11],
       [ 3, 73]])

*CM is giving the value of true +VE false -ve ,false+ve and true -ve respt.*

In [40]:
print(classification_report(Y_test,Y_predict))

              precision    recall  f1-score   support

           0       0.90      0.71      0.79        38
           1       0.87      0.96      0.91        76

    accuracy                           0.88       114
   macro avg       0.88      0.84      0.85       114
weighted avg       0.88      0.88      0.87       114



Model is giving the accuracy of 88% for classification of cancer tumor type either malignant or benign.

